### ISSUE TITLE
0.6.0.0-alpha Modded Container Issue

### ISSUE THREAD

**Issue by AnonymousXXI** [TheXFactor117/Loot-Slash-Conquer-Pre1.14#2]
Love the mod and I'm glad you guys released it. With this mod on I am unable to interact with any modded chest or furnace.

ironchest-1.12.2-7.0.40.824
CookingForBlockheads_1.12.2-6.4.40

The chest/furnace menus for these items does not appear.

**Comment by TheXFactor117** [comment]
I'll look into it - thanks for the report!

**Comment by TheXFactor117** [comment]
I'm able to open the chests, put stuff in, take stuff out, and see them even in Crystal Chests.

![](https://user-images.githubusercontent.com/6678711/42142973-8aed6d6e-7d80-11e8-98f0-6754e7ae5687.png)

![](https://user-images.githubusercontent.com/6678711/42142982-998d9f6a-7d80-11e8-8d3b-c70458597942.png)

Does the issue only happen when Loot Slash Conquer is loaded? In other words, does the issue go away if you remove just Loot Slash Conquer? If that is the case, perhaps there's some other context I'm missing.

**Comment by AnonymousXXI** [comment]
Yeah, the issue stopped when I removed the mod. Maybe it's because I loaded the mod onto a world that was already generated. I'll put it back in and see if it's a compatibility issue with another mod I have.

**Comment by TheXFactor117** [comment]
I don't really see why Loot Slash Conquer would affect this - though some issues can be like that. If you can reproduce it easily, let me know and I can try and reproduce it as well.

**Comment by AnonymousXXI** [comment]
It works just fine in single-player. I completely forgot to mention that I'm playing on a server with a friend.

**Comment by AnonymousXXI** [comment]
also found this in the server log

[00:11:53] [Server thread/ERROR] [FML]: 0: NORMAL
[00:11:53] [Server thread/ERROR] [FML]: 1: ASM: class com.lsc.events.EventContainerOpen onContainerOpen(Lnet/minecraftforge/event/entity/player/PlayerContainerEvent;)V

**Comment by TheXFactor117** [comment]
That’ll do it. I’ll get that hotfixed tomorrow along with some other bugs and tweaks. Hang tight!

**Comment by TheXFactor117** [comment]
I tested this in my dev environment and wasn't able to reproduce it. When you do it, is there a bigger log error that appears? Also, does it never work for any chests or does it not work for certain circumstances?

**Comment by AnonymousXXI** [comment]
[14:20:20] [Server thread/ERROR] [FML]: Exception caught during firing event net.minecraftforge.event.entity.player.PlayerContainerEvent$Close@74774532:
java.lang.NoSuchMethodError: net.minecraft.item.ItemArmor.func_185083_B_()Lnet/minecraft/inventory/EntityEquipmentSlot;
	at com.lsc.loot.generation.ItemGeneratorHelper.setTypes(ItemGeneratorHelper.java:56) ~[ItemGeneratorHelper.class:?]
	at com.lsc.loot.generation.ItemGenerator.create(ItemGenerator.java:29) ~[ItemGenerator.class:?]
	at com.lsc.events.EventContainerOpen.generate(EventContainerOpen.java:76) ~[EventContainerOpen.class:?]
	at com.lsc.events.EventContainerOpen.onContainerOpen(EventContainerOpen.java:53) ~[EventContainerOpen.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_198_EventContainerOpen_onContainerOpen_PlayerContainerEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:179) [EventBus.class:?]
	at net.minecraft.entity.player.EntityPlayerMP.func_71128_l(EntityPlayerMP.java:1023) [oq.class:?]
	at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.openGui(FMLNetworkHandler.java:92) [FMLNetworkHandler.class:?]
	at net.minecraft.entity.player.EntityPlayer.openGui(EntityPlayer.java:2550) [aed.class:?]
	at cpw.mods.ironchest.common.blocks.chest.BlockIronChest.func_180639_a(BlockIronChest.java:121) [BlockIronChest.class:?]
	at net.minecraft.server.management.PlayerInteractionManager.func_187251_a(PlayerInteractionManager.java:446) [or.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_184337_a(NetHandlerPlayServer.java:739) [pa.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:55) [ma.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:11) [ma.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_144]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_144]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_144]
[14:20:20] [Server thread/ERROR] [FML]: Index: 1 Listeners:
[14:20:20] [Server thread/ERROR] [FML]: 0: NORMAL
[14:20:20] [Server thread/ERROR] [FML]: 1: ASM: class com.lsc.events.EventContainerOpen onContainerOpen(Lnet/minecraftforge/event/entity/player/PlayerContainerEvent;)V


When this happens the only thing I can open are the chest from vanilla Minecraft. I can't interact with anything else from other mods, when I try to use a keypad from SecurityCraft the window doesnt open. For modded chest the animation for them opening plays but no menu for the chest pops up.

**Comment by TheXFactor117** [comment]
Alright - it’s definitely an issue. I’ll try and reproduce but it looks like it’s trying to call a piece of client side code.

Hopefully I’ll find a fix for it soon.

**Comment by TheXFactor117** [comment]
I've attached a .zip which contains a testing build of the mod which may fix the problem. I can't reproduce inside of my dev environment, and I just figured it would be easier to give you a testing jar to see if the issue is fixed instead of me having to create a forge server and run it out of my environment.

I figured out what's causing the issue, and made some changes which may fix it. I'm not 100% sure if it will be I think it should work. If not, I'll have another idea as to how to fix it.

[LSC - Issue #2.zip](https://github.com/TheXFactor117/Loot-Slash-Conquer/files/2157335/LSC.-.Issue.2.zip)
